WordPress/Classes/ViewRelated/Notifications/Views/NotificationDetailUserView.swift

```diff
@@ -0,0 +1,174 @@
+
+struct NotificationDetailUserView: View {
+	typealias FollowConfig = NotificationDetailUserView.FollowActionConfiguration
+
+    let config: Configuration
+
+    init(avatarURL: URL?, username: String?, blog: String?, onUserClicked: @escaping () -> Void) {
+        self.config = Configuration(avatarURL: avatarURL, username: username, blog: blog, onUserClicked: onUserClicked)
+    }
+
+    init(
+        avatarURL: URL?,
+        username: String?,
+        blog: String?,
+        isFollowed: Bool,
+        onUserClicked: @escaping () -> Void,
+        onFollowClicked: @escaping (Bool) -> Void
+    ) {
+        self.config = Configuration(
+            avatarURL: avatarURL,
+            username: username,
+            blog: blog,
+            followActionConfig: FollowConfig(isFollowed: isFollowed, onFollowClicked: onFollowClicked),
+            onUserClicked: onUserClicked
+        )
+    }
+
+    var body: some View {
+        HStack {
+            Button(action: config.onUserClicked) {
+                HStack(spacing: .DS.Padding.split) {
+                    AvatarsView(style: .single(config.avatarURL))
+                    VStack(alignment: .leading) {
+                        if let username = config.username {
+                            Text(username)
+                                .style(.bodySmall(.regular))
+                                .foregroundStyle(Color.DS.Foreground.primary)
+                                .lineLimit(1)
+                        }
+                        if let blog = config.blog {
+                            Text(blog)
+                                .style(.bodySmall(.regular))
+                                .foregroundStyle(Color.DS.Foreground.secondary)
+                                .lineLimit(1)
+                        }
+                    }
+                    Spacer()
+                }
+            }
+            if let followAction = config.followActionConfig {
+                FollowButton(config: followAction)
+            }
+        }
+        .padding(.horizontal, .DS.Padding.double)
+        .padding(.top, .DS.Padding.double)
+    }
+
+    struct Configuration {
+        let avatarURL: URL?
+        let username: String?
+        let blog: String?
+        let followActionConfig: FollowActionConfiguration?
+        let onUserClicked: () -> Void
+
+        init(
+            avatarURL: URL?,
+            username: String?,
+            blog: String?,
+            followActionConfig: FollowActionConfiguration? = nil,
+            onUserClicked: @escaping () -> Void
+        ) {
+            self.avatarURL = avatarURL
+            self.username = username
+            self.blog = blog
+            self.followActionConfig = followActionConfig
+            self.onUserClicked = onUserClicked
+        }
+    }
+
+    struct FollowActionConfiguration {
+        let isFollowed: Bool
+        let onFollowClicked: (Bool) -> Void
+    }
+
+}
+
+private struct FollowButton: View {
```

Comment:
Can we not use `DSButton` here instead of defining a new button? Tertiary style with small size

Reply:
Yeah, great idea! Thanks for double checking it with Chris.
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/23178/commits/fc41a45cb5d90b1ca4dc8c159488ad180d6a5b25

I haven't changed the titles. I would proceed with renaming "Follows" to "Subscribers" feature in general within a separate task/PR. Will prepare an issue for it.